core/services/run_manager.go

```diff
@@ -133,23 +103,16 @@ func NewRun(
 		}
 
 		runAdapters = append(runAdapters, adapter)
-
-		minimumConfirmations := clnull.Uint32{}
-		if currentHeight != nil {
-			minimumConfirmations = clnull.Uint32From(
-				utils.MaxUint32(
-					config.MinIncomingConfirmations(),
-					task.Confirmations.Uint32,
-					adapter.MinConfs()),
-			)
+		if currentHeight == nil {
+			continue
 		}
 
-		run.TaskRuns[i] = models.TaskRun{
-			ID:                   models.NewID(),
-			JobRunID:             run.ID,
-			TaskSpec:             task,
-			MinimumConfirmations: minimumConfirmations,
-		}
+		run.TaskRuns[i].MinimumConfirmations = clnull.Uint32From(
```

Comment:
This still appears to be a change in behaviour. Why aren't we creating a new ID any more? Or, alternatively, why were we creating a new ID before?

Reply:
We are creating a new ID in MakeJobRun.

Reply:
Added a test to demonstrate.